packages/static-config/src/swc.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  ArrayExpression,
+  BooleanLiteral,
+  ExportDeclaration,
+  Identifier,
+  KeyValueProperty,
+  Module,
+  Node,
+  NullLiteral,
+  NumericLiteral,
+  ObjectExpression,
+  StringLiteral,
+  VariableDeclaration,
+} from '@swc/core';
+import { FromSchema, JSONSchema } from 'json-schema-to-ts';
+import { validate } from './validation';
+
+function isExportDeclaration(node: Node): node is ExportDeclaration {
+  return node.type === 'ExportDeclaration';
+}
+
+function isVariableDeclaration(node: Node): node is VariableDeclaration {
+  return node.type === 'VariableDeclaration';
+}
+
+function isIdentifier(node: Node): node is Identifier {
+  return node.type === 'Identifier';
+}
+
+function isBooleanLiteral(node: Node): node is BooleanLiteral {
+  return node.type === 'BooleanLiteral';
+}
+
+function isNullLiteral(node: Node): node is NullLiteral {
+  return node.type === 'NullLiteral';
+}
+
+function isStringLiteral(node: Node): node is StringLiteral {
+  return node.type === 'StringLiteral';
+}
+
+function isNumericLiteral(node: Node): node is NumericLiteral {
+  return node.type === 'NumericLiteral';
+}
+
+function isArrayExpression(node: Node): node is ArrayExpression {
+  return node.type === 'ArrayExpression';
+}
+
+function isObjectExpression(node: Node): node is ObjectExpression {
+  return node.type === 'ObjectExpression';
+}
+
+function isKeyValueProperty(node: Node): node is KeyValueProperty {
+  return node.type === 'KeyValueProperty';
+}
```

Comment:
Seems like the types are not defined properly if these assertions are necessary? I would expect `if (node.type === 'KeyValueProperty')` to narrow the type automatically 🤔

Reply:
It wish so but it doesn't. tsc cannot infer `KeyValueProperty` from something like `if (node.type === 'KeyValueProperty')` and `node` is still `Node` within the then block.

JFYI, some [suggestions](https://github.com/microsoft/TypeScript/issues/10421) are in proposed but seems not active.

Reply:
In this case, I don't think the issue is TS.

Its likely the `@swc/core` types are wrong. It just says string here 

Can we fix it upstream in swc?

<img width="308" alt="image" src="https://user-images.githubusercontent.com/229881/170553272-0484bf17-d457-4a48-8058-0a417d652284.png">